test/python/test_rticonnextdds_discovery.py

```diff
@@ -0,0 +1,195 @@
+###############################################################################
+# (c) 2019 Copyright, Real-Time Innovations.  All rights reserved.            #
+# No duplications, whole or partial, manual or electronic, may be made        #
+# without express written permission.  Any such copies, or revisions thereof, #
+# must display this notice unaltered.                                         #
+# This code contains trade secrets of Real-Time Innovations, Inc.             #
+###############################################################################
+
+import pytest,time,sys,os
```

Comment:
Are you testing a publication/subscription that doesn't set the entity name Qos?

Reply:
Added a test for this

Reply:
Following up on this...
I can't test an actual xsi:nil="true" entity name policy since connector_new will fail due to an invalid XML file.
The name tag in  `<data_reader name="the_name" topic_ref="the_topic"/> ` sets the entity_name and has to be present. I will add a test using an empty string for the name.

Reply:
It should be possible to test this in the binding though